docs: align readme rpc docs

Mirko-von-Leipzig opened the pull request:
This PR re-aligns the various RPC docs with the proto files.

I've opted to only provide a summary and not a full parameter doxygen style documentation as there was before as I don't think its trivial to keep aligned with reality, nor sufficient -- users will always have to use and/or read the proto files.

Closes #626

igamigo commented on crates/rpc/README.md:
```diff
@@ -1,129 +1,112 @@
+### SubmitProvenTransaction
 
-- `notes`: `[Note]` - List of notes matching the list of requested NoteIds.
+Submits proven transaction to the Miden network.
```
I know this might have been deliberately left aside (and/or it might not be worth mentioning here) but I think it could be good to mention that this returns the current block height

Mirko-von-Leipzig replied:
We can add it in; tbh I find these docs low-value as I sort of mention in [this discussion](https://github.com/0xPolygonMiden/miden-node/discussions/630). They're neither automatically sync'd, nor fully descriptive so they end up just defaulting to "meh".

bobbinth commented on crates/rpc/README.md:
```diff
@@ -1,129 +1,112 @@
+Note synchronization request.
 
-- `account_id`: `AccountId` – account ID.
+Specifies note tags that client is interested in. The server will return the first block which contains a note matching
+`note_tags` or the chain tip.
```
I think the second sentence should be more like: "The server will return the matching notes from the first block which contains a note matching `note_tags`."

I would also maybe make this description more detailed - similar to how we have it for `SyncState` endpoint (and probably propagate these details to the protobuf files as well).

bobbinth commented on crates/store/README.md:
```diff
@@ -1,163 +1,133 @@
 
 ### ApplyBlock
 
 Applies changes of a new block to the DB and in-memory data structures.
```
I would also maybe mention that this also saves the raw block data as a flat file.

bobbinth commented on crates/store/README.md:
```diff
@@ -1,163 +1,133 @@
+### CheckNullifiers
 
-**Returns**
+Gets a list of proofs for given nullifier hashes, each proof as a sparse Merkle Tree.
```
Same comment as for the analogous endpoint for the RPC.

bobbinth commented on crates/store/README.md:
```diff
@@ -1,163 +1,133 @@
 
 ### GetBlockInputs
 
 Returns data required to prove the next block.
```
For this and similar endpoints, should we mention that they are meant to be used by the block producer?

bobbinth commented on crates/store/README.md:
```diff
@@ -1,163 +1,133 @@
-
-For preserving some degree of privacy, note tags and nullifiers filters contain only high part of hashes. Thus, returned data
-contains excessive notes and nullifiers, client can make additional filtering of that data on its side.
+Returns info which can be used by the client to sync up to the latest state of the chain for the objects (accounts,
```
Same comment as for the analogous endpoint for the RPC.

bobbinth commented on crates/store/README.md:
```diff
@@ -1,163 +1,133 @@
+### CheckNullifiersByPrefix
 
-Get a list of proofs for given nullifier hashes, each proof as a sparse Merkle Tree
+Returns a list of nullifiers that match the specified prefixes and are recorded in the node.
```
Same comment as for the analogous endpoint for the RPC.

bobbinth commented on crates/store/README.md:
```diff
@@ -1,163 +1,133 @@
 
 ### GetBlockByNumber
 
 Retrieves block data by given block number.
```
Same comment as for the analogous endpoint for the RPC.

bobbinth reviewed (approved):
Looks good! Thank you! I left some small comments inline.

Mirko-von-Leipzig commented:
I've addressed all comments I believe. In general there are probably still discrepancies between the proto service/request/response/type comments and the readme's.

imo we should only "publicly" document the RPC API as its the only real external one; and leave the internal APIs undocumented, at least for now. Devs can use the proto files and code docs.

bobbinth reviewed (approved):
Looks good! Thank you!

> imo we should only "publicly" document the RPC API as its the only real external one; and leave the internal APIs undocumented, at least for now. Devs can use the proto files and code docs.

Yes, agree with this. We should move toward this state.
